src/modules/fw_mode_manager/FixedWingModeManager.cpp

```diff
@@ -1263,6 +1263,75 @@ FixedWingModeManager::control_auto_takeoff(const hrt_abstime &now, const float c
 	}
 }
 
+void
+FixedWingModeManager::control_auto_takeoff_no_nav(const hrt_abstime &now, const float control_interval,
+		const float current_setpoint_altitude_amsl)
+{
+	if (!_control_mode.flag_armed) {
+		reset_takeoff_state();
+	}
+
+	const float takeoff_airspeed = (_param_fw_tko_airspd.get() > FLT_EPSILON) ? _param_fw_tko_airspd.get() :
+				       _param_fw_airspd_min.get();
+
+	if (_runway_takeoff.runwayTakeoffEnabled()) {
+		// TODO runway takeoff without nav
+
+	} else {
+		/* Perform launch detection */
+		if (!_skipping_takeoff_detection && _param_fw_laun_detcn_on.get() &&
+		    _launchDetector.getLaunchDetected() < launch_detection_status_s::STATE_FLYING) {
+
+			if (_control_mode.flag_armed) {
+				/* Detect launch using body X (forward) acceleration */
+				_launchDetector.update(control_interval, _body_acceleration_x); // TODO: check if using lpos.a is okay
+			}
+
+		} else	{
+			/* no takeoff detection --> fly */
+			_launchDetector.forceSetFlyState();
+		}
+
+		if (!_launch_detected && _launchDetector.getLaunchDetected() > launch_detection_status_s::STATE_WAITING_FOR_LAUNCH) {
+			_launch_detected = true;
+			_takeoff_ground_alt = _current_altitude;
+		}
+
+		/* Launch has been detected, hence we have to control the plane. */
+
+		fixed_wing_lateral_setpoint_s fw_lateral_ctrl_sp{empty_lateral_control_setpoint};
+		fw_lateral_ctrl_sp.timestamp = now;
+		fw_lateral_ctrl_sp.lateral_acceleration = 0.f; // level wings
+
+		_lateral_ctrl_sp_pub.publish(fw_lateral_ctrl_sp);
+
+		const float max_takeoff_throttle = (_launchDetector.getLaunchDetected() < launch_detection_status_s::STATE_FLYING) ?
+						   _param_fw_thr_idle.get() : NAN;
+		const fixed_wing_longitudinal_setpoint_s fw_longitudinal_control_sp = {
+			.timestamp = now,
+			.altitude = NAN,
+			.height_rate = 3.f, //TODO get from param? Or just full weight on airspeed with throttle constraints?
```

Comment:
@sfuhrer In the normal takeoff function, you have used the max climb rate. Why not use the same here?